[이진우] sprint9

ajantang opened the pull request:
## 요구사항

### 공통

- [x] Github에 위클리 미션 PR을 만들어 주세요.
- [x] React 혹은 Next.js를 사용해 진행합니다.
- [x] 프론트엔드에서 api 요청 구현은 TanStack React Query를 활용해 주세요.
- [x] API는 https://panda-market-api.vercel.app를 사용합니다. 명세는 https://panda-market-api.vercel.app/docs를 확인해 주세요.

### 로그인/회원가입 페이지

- [x] JavaScript로 구현한 로그인/회원가입 페이지를 React.js 혹은 Next.js로 마이그레이션해 주세요.

### 로그인 페이지

- [x] "회원 가입하기"를 클릭하면 회원가입 페이지로 이동해 주세요.
- [x] 로그인 실패하는 경우, 이메일 input 아래에 "이메일을 확인해 주세요.", 비밀번호 input 아래에 "비밀번호를 확인해 주세요." 에러 메시지를 표시해 주세요.
- [x] 로그인 버튼이 활성화된 후, 로그인 버튼 클릭 또는 Enter키 입력으로 로그인 실행합니다.
- [x] "/api/signIn"으로 POST 요청해서 성공 응답을 받으면 중고 마켓 페이지로 이동합니다. 참고로 JWT로 구현되어 있습니다.
- [x] 실패할 경우, 실패 메시지를 모달을 통해 표시합니다.

### 회원가입 페이지

- [x] "회원 가입하기"를 클릭하면 '/signin' 페이지로 이동합니다
- [x] 회원가입 버튼 클릭 또는 Enter키 입력으로 회원가입을 실행합니다.
- [x] 비밀번호 input과 비밀번호 확인 input의 값이 다른 경우, 비밀번호 확인 input 아래에 "비밀번호가 일치하지 않아요." 에러 메시지를 표시해 주세요.
- [x] 버튼이 활성화된 후, 회원가입은 "/api/signUp" POST 요청해서 진행합니다. 참고로 JWT로 구현되어 있습니다.
- [x] 회원가입 성공 응답을 받으면 중고마켓 페이지로 이동합니다.
- [x] 실패할 경우, 실패 메시지를 모달을 통해 표시합니다.

### 로그인, 회원가입 페이지 공통

- [x] 눈 모양 아이콘 클릭 시 비밀번호의 문자열이 보이기도 하고, 가려집니다.
- [x] 비밀번호의 문자열이 가려질 때는 눈 모양 아이콘에는 사선이 그어져 있고, 비밀번호의 문자열이 보일 때는 사선이 없는 눈 모양 아이콘이 보입니다.
- [x] 소셜 로그인에 구글 아이콘 클릭 시 'https://www.google.com', 카카오 아이콘 클릭 시 'https://www.kakaocorp.com/page'로 이동합니다.
- [x] 로그인/회원가입 시 성공 응답으로 받은 accessToken을 로컬 스토리지에 저장합니다.
- [x] 로그인/회원가입 페이지에 접근 시 로컬 스토리지에 accessToken이 있는 경우 '/folder' 페이지로 이동합니다.

### GNB

- [x] 상단 내비게이션 바에 프로필 영역은 인가된 경우, 유저 정보 API를 활용해 주세요.
- [x] 인가되지 않았을 경우 "로그인" 버튼이 보이게 해 주세요.

### 상품 상세 페이지

- [x] PC, Tablet, Mobile 디자인에 해당하는 상품 상세 페이지를 만들어 주세요.
- [x] 상품 상세 페이지 url path는 "/items/{itemId)"로 설정하세요.
- [x] '목록으로 돌아가기' 버튼 클릭 시 중고마켓 페이지 "/items"로 이동합니다.
- [x] 상품 상세 데이터는 '/products/{productId}' GET 메서드 사용해 불러오세요. 이때, 상품 상세 조회는 인가된 사용자만 이용할 수 있도록 합니다.
  - [x] 상품에 대한 댓글 조회도 가능합니다.
- [x] 상품 수정 및 삭제 기능을 API를 활용해 구현합니다. 이때, 인가된 사용자만 이용할 수 있도록 합니다.
  - [x] 상품 수정은 '/products/{productId}' PUT을 사용합니다.
  - [x] 상품 삭제는 '/products/{productId}' DELETE를 사용합니다.
- [x] 상품 삭제 전, 확인 모달을 띄워주세요.
- [x] 상품에 대한 좋아요 및 좋아요 취소 기능을 https://panda-market-api.vercel.app/docs에 명세된 '/products/{productId}/favorite' POST & DELETE 활용해 구현합니다. 이때 인가된 사용자만 좋아요 기능을 이용할 수 있도록 합니다.
- [x] 댓글 생성 및 삭제 기능을 API를 활용해 구현합니다. 이때, 인가된 사용자만 이용할 수 있도록 합니다.
  - [x] 댓글 수정은 https://panda-market-api.vercel.app/docs에 명세된 '/products/{productId}/comments' PUT을 사용합니다.
  - [x] 댓글 삭제는 https://panda-market-api.vercel.app/docs에 명세된 '/products/{productId}/comments' DELETE를 사용합니다.

## 심화 요구사항
### 로그인 및 회원가입 페이지 공통

- [x] 로그인, 회원가입 기능에 react-hook-form을 활용해 주세요.
- [x] 브라우저에 현재 보이는 화면의 영역(viewport) 너비를 기준으로 분기되는 반응형 디자인을 적용합니다.
  - PC: 1200px 이상
  - Tablet: 744px 이상 ~ 1199px 이하
  - Mobile: 375px 이상 ~ 743px 이하
  - 375px 미만 사이즈의 디자인은 고려하지 않습니다

### 유저 기능

- [x] 리퀘스트 헤더에 인증 토큰을 첨부할 때 axios interceptors를 활용해 주세요. (axios를 사용하지 않는다면 이와 유사한 기능을 활용해 주세요.)

### React-Query로 마이그레이션

- [x] fetch 혹은 axios로 구현된 기존의 API 요청 코드를 React-Qeury로 마이그레이션 합니다.

### 로딩 및 에러 핸들링

- [x] 로딩 인디케이터와 에러 메시지를 구현합니다.
- [x] 상품 목록 및 상품 상세 데이터를 Prefetching 합니다

### 상품 데이터 캐싱 및 업데이트

- [x] React Query의 캐싱 기능을 활용하여 데이터 로딩 시간을 최소화합니다.
- [x] 상품 목록 페이지에서 데이터의 실시간 업데이트를 위해 적절한 Query Refresh 설정을 적용합니다.

## 주요 변경사항

- 미구현 요구사항 구현
- 배포 : https://sprint-fe-snowrangs-projects.vercel.app/

## 스크린샷

로그인 - 모달
![image](https://ifh.cc/g/gKLva7.png)
회원가입 - 모달
![image](https://ifh.cc/g/jrtPqc.png)
중고마켓 페이지
![image](https://ifh.cc/g/Nfjk64.jpg)
상품 상세 페이지
![image](https://ifh.cc/g/kaG37L.jpg)
댓글 수정
![image](https://ifh.cc/g/ABrVTm.png)
상품 삭제 모달
![image](https://ifh.cc/g/9Zylsa.jpg)

## 멘토에게

- 조금 급하게 작업하느라 멘토님의 피드백을 다 확인 못하고 작업을 진행했습니다. 추후 반영하도록 하겠습니다.
- useAuth 부분의 login 쪽에 혹시 문제가 있는지 피드백 부탁드립니다(로그인 상태에서 상품 상세 페이지 직접 접근시 로그인 상태 정보를 잘 못 읽어 오는거 같습니다(AuthRoute 컴포넌트))
- 상품 삭제하는 과정에서 DropDownKebabProduct에서 modal 창까지 넣어서 구현했는데, 아무래도 modal창을 kebab 안에 넣는건 안 좋아 보이는데, 멘토님 의견이 궁금합니다
- 아직 일부 기능의 테스트가 마무리되지 않았습니다
- 클린 코드를 위해 일부 코드 수정 중입니다 
- 테스트가 용이하도록 vercel에 임시로 배포했습니다 https://sprint-fe-snowrangs-projects.vercel.app/
- 추후에 확인한거지만, 배포한 페이지에 동적으로 생성되는 경로의 페이지가 404가 뜨는데, 원인을 알 수 없습니다. 아마도 vercel 쪽 설정 문제로 추정되는데(콘솔에서는 문제없었습니다) 혹시 정보가 있으시면 알려주시길 바랍니다(vercel의 빌드 로그와 프로젝트에 선언한 vercel.json 파일 사진을 아래 첨부합니다)

빌드 로그
![image](https://ifh.cc/g/x5GYta.png)
vercel.json
![image](https://ifh.cc/g/92OAyw.png)

ajantang committed:
style : apply header component tailwind css by using classNames lib

header에 적용하는 css classNames 라이브러리를 사용하여 최대한 tailwind css 형식으로 적용

ajantang committed:
style : apply tailwind css style in nav, footer, dropdown component

tailwind 스타일에 맞춰 css 수정

ajantang committed:
style : apply tailwind css in header-user component

로직 추가 전 tailwind css 스타일 적용

ajantang committed:
style : divide client component & apply useQuery, tailwind in ArticleDetailClient component

client component 세분화
useQuery 적용
tailwind css 적용

참고 : 미션9를 우선 진행하기 위해, page.js / ArticleDetailComponet 통합 & component 일부 css tailwind 적용 보류

ajantang committed:
feat : add api function for codeit api & apply user api in HeaderUser

미션용 codeit api 추가
HeaderUser에 user api를 사용해 로그인 상태 변환

참고 : refresh token을 쿠키에서 사용할 수 없어(codeit에서 허용 안함) 임시로 로컬에 저장

ajantang committed:
feat : apply app router/tailwind css in flea-market page(unfinished)

상품 상세 페이지 접근을 위한 중고 마켓 페이지 마이그레이션
401 문제로 로그인/회원가입 구현을 위해 마이그레이션 마무리 못함

ajantang committed:
add base file of sign-up, sign-in

ajantang committed:
feat : install react-hook-form

ajantang committed:
feat : add sign-in code(unfinished)

API 연결 및 visible 버튼 css 수정 필요
테스트 필요

ajantang committed:
feat : add sign-in code

api 적용 & 디자인 적용

참고 : sign-in 완료 후 홈 자동으로 이동되는 홈화면의 HeaderUser의 로그인 상태 갱신이 안되는 현상 테스트 필요

ajantang committed:
feat : set sign-in state by using zustand

로그인 상태 관리를 위해 zustand를 사용

ajantang committed:
feat : add sign-in & sign-up code

간편 로그인 추가

참고 : sign-up 미완성

ajantang committed:
feat : add sign-up & manage user info by using zustand

회원 가입 페이지 코드 추가
회원 가입/로그인 과정에서 받은 사용자 정보 관리 코드 추가

ajantang committed:
test : add server component test code

상품 조회 관련 api 사용 중 localstorage에 저장된 token 정보를 못 가져와 문제가 발생하는 것을 확인
결국 client로 내려야 되는지 대안 확인 중

참고 : 아직 수정 안함

wildCodingWarrior commented on app/bulletin-board/BestBoard.js:
```diff
@@ -1,30 +1,51 @@
-  const boardLabelClass = `font-bold ${style.label}`;
-  const boardListClass = `flex flex-row ${style.list}`;
+export async function BestBoard() {
+  const boardClass = classNames(
+    "w-pc-content",
+    "h-pc-best-board",
+    "tablet:w-tablet-content",
+    "tablet:h-tablet-best-board",
+    "mobile:w-mobile-content",
+    "mobile:h-mobile-best-board"
+  );
+  const boardLabelClass = classNames(
+    "h-2.4rem",
+    "text-xl",
+    "leading-32",
+    "font-bold",
+    "mobile:h-2.6rem",
+    "mobile:text-2lg",
+    "mobile:leading-26"
+  );
+  const boardListClass = classNames(
+    "flex",
+    "flex-row",
+    "mt-2.4rem",
+    "w-pc-content",
+    "h-pc-best-board-list",
+    "gap-2.4rem",
+    "overflow-hidden",
+    "tablet:w-tablet-content",
+    "tablet:h-tablet-best-board-list",
+    "gap-1.6rem",
+    "mobile:w-mobile-content",
+    "mobile:h-mobile-best-board-list",
+    "mobile:mt-1.6rem"
+  );
```
관리하기에 이게 더 좋네요. classNames 라이브러리

wildCodingWarrior commented on app/bulletin-board/BestBoard.js:
```diff
@@ -34,9 +55,10 @@ export function BestBoard() {
+
+      return newList;
+    }
+  );
```
P3
await와 then을 혼용하고 있습니다. 일관성을 위해 하나의 방식만 사용하는 것이 좋습니다.

wildCodingWarrior replied:
P3
map 함수에서 index를 key로 사용하는 것은 권장되지 않으니, 대신 고유한 id를 사용하세요.

ajantang replied:
then을 선호하는 편입니다. 어느 순간 await를 쓰는 부분이 있는데, 이 부분은 수정해두겠습니다

wildCodingWarrior commented on app/bulletin-board/Board.js:
```diff
@@ -13,120 +15,105 @@ import {
-      return;
-    }
-    getArticles(page + 1, PAGE_SIZE_BY_DEVICE[device], recentOrder).then(
-      (data) => {
-        const newList = data.articles.map((article, index) => {
-          return getArticleTag(article, index);
-        });
-
-        setPage(page + 1);
-        setList([...list, newList]);
-        setAddingList(false);
+      if (isBottom && hasNextPage && !isFetchingNextPage) {
+        fetchNextPage();
       }
-    );
-  }, [addingList]);
+    }, 200);
+
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, [hasNextPage, isFetchingNextPage, fetchNextPage]);
 
```
P4
보통 스크롤 위치로 판별하기보다는 intersection observer를 많이 사용합니다. 한 번 알아보시는 걸 권장드립니다.

wildCodingWarrior commented on app/components/CommentSection.js:
```diff
@@ -0,0 +1,48 @@
+  });
+
+  const mutation = useMutation({
+    mutationFn: ({ postId, content }) => createPostComment(postId, content),
```
P3
사사롭지만 타입스크립트까지 적용하게 될 무렵에는 {postId, content} 이런식으로 파라미터를 객체로 넣는 방식을 권장합니다.

wildCodingWarrior commented on app/components/EmailInput.js:
```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import classNames from "classnames";
+
+export default function EmailInput({ label, register, errors }) {
+  let inputClass = classNames("sign-in__input", "focus:border-input--focus");
+  if (errors.password) {
+    inputClass = classNames(
+      "sign-in__input",
+      "focus:border-input--focus",
+      "invalid-border"
+    );
+  }
+
+  return (
+    <div className="sign-in__input-set">
+      <label className="sign-in__label" htmlFor="email">
+        이메일
+      </label>
+      <input
+        className={inputClass}
+        id="email"
+        type="email"
+        placeholder="이메일을 입력해주세요"
+        {...register(label, {
+          required: "이메일을 입력해주세요",
+          pattern: {
+            value: /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/,
+            message: "이메일 형식이 아닙니다",
+          },
+        })}
+        aria-invalid={errors.email ? "true" : "false"}
+      />
+      {errors.email && <p className="sign-in__warn">{errors.email.message}</p>}
+    </div>
+  );
+}
```
P4
전체적으로 form을 구현하기 쉽지 않은 부분 중 하나인데, 이 부분 나중에 react-hook-form 한번 알아보시면 좋을 것 같아요.

wildCodingWarrior replied:
아 이미 활용하셨군요

wildCodingWarrior commented on app/components/HeaderUser.js:
```diff
@@ -0,0 +1,74 @@
+    alert("아직 구현되지 않은 기능입니다");
+  };
+
+  const result = isSignedIn ? (
+    <div className={userInfoClass} onClick={handleProfileClick}>
+      <Profile
+        type={PROFILE_H40}
+        profileImgUrl={userProfileUrl}
+        onClick={handleProfileClick}
+      />
+      <p className={userNicknameClass}>{userNickname}</p>
+    </div>
+  ) : (
+    <BtnLogin />
+  );
```
P2
가독성 측면에서 result로 만들기보다는 JSX return 하는 곳에서 곧바로 사용하는게 더 좋겠습니다.

ajantang replied:
수정 완료

wildCodingWarrior commented on app/items/[productId]/page--client.js:
```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect } from "react";
+import { getProduct } from "@/lib/api-codeit-product";
+
+export default function ProductPage({ params }) {
+  const { productId } = params;
+  const [userId, setUserId] = useState("");
+  const [productName, setProductName] = useState("");
+  const [description, setDescription] = useState("");
+  const [price, setPrice] = useState("0");
+  const [tags, setTags] = useState([]);
+  const [images, setImages] = useState([]);
+  const [ownerId, setOwnerId] = useState("");
+  const [favoriteCount, setFavoriteCount] = useState(0);
+  const [createdDate, setCreatedDate] = useState("2024-09-25T04:49:03.316Z");
+  const [ownerNickname, setOwnerNickname] = useState("");
+  const [isFavorite, setIsFavorite] = useState("");
+
+  useEffect(() => {
+    getProduct({ productId: productId })
+      .then((data) => {
+        setUserId(data.id);
+        setProductName(data.name);
+        setDescription(data.description);
+        setPrice(data.price);
+        setTags(data.tags);
+        setImages(data.images);
+        setOwnerId(data.ownerId);
+        setFavoriteCount(data.favoriteCount);
+        setCreatedDate(data.createdDate);
+        setOwnerNickname(data.ownerNickname);
+        setIsFavorite(data.isFavorite);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }, []);
```
따로 state 여러개 안 만들고 하나의 product 객체를 담는 state를 만들면 어떨까요?

wildCodingWarrior commented on lib/api-post.js:
```diff
@@ -0,0 +1,72 @@
+  }
+}
+
+export async function getPosts(page, pageSize, orderBy, keyword) {
+  const headers = { authorization: userId }; // 임시로 사용
+  const path = "/article";
+  const params = {
+    ...(page && { page }),
+    ...(pageSize && { pageSize }),
+    ...(orderBy && { orderBy }),
+    ...(keyword && { keyword }),
+  };
+
```
아주 좋은 테크닉이네요.

wildCodingWarrior commented on lib/api-post.js:
```diff
@@ -0,0 +1,72 @@
+  }
+}
+
+export async function getPosts(page, pageSize, orderBy, keyword) {
```
P3
이런식으로 파라미터가 여러개 있는 곳은 항상 객체로 params 받아주는게 좋습니다.

wildCodingWarrior reviewed (changes requested):
고생하셨습니다 ! 점점 나아지는게 느껴지네요. 커스텀 훅만 더 활용 잘 해봅시다.

ajantang committed:
fix & feat : fix items page & add items detail page

중고 마켓 페이지 오류 수정
상품 상세 페이지 추가(댓글 기능 추가 예정)

ajantang committed:
feat : add code of product page, product comment maker/list component

상품 페이지의 댓글 관련 컴포넌트 추가 & 수정
상품 댓글 관련 api 추가 & 연결

참고 : 상품/상품 댓글 수정/삭제 추가 예정

ajantang committed:
feat : add sign in fail message

요구사항에 맞춰 로그인 실패시 에러 메세지 출력

ajantang committed:
feat : add sign-in modal

로그인 실패에 모달창 추가
일반 팝업창으로 만든거라, react-modal을 사용할 예정

ajantang committed:
feat : replace modal by using react-modal

react-modal을 사용하여 modal 기능 교체

ajantang committed:
feat : add modal in sign up

요구사항에 맞춰 회원 가입 모달 추가
요구사항에 맞춰 비밀번호 visible image 변경

ajantang committed:
feat : add redirect route in sign in

요구사항에 맞춰 로그인 상태에서 sign-in 페이지 접근시 중고마켓 페이지로 이동

참고 : 요구사항에 /folder 경로 이동을 원했지만, 경로에 페이지가 없어 /items 페이지로 이동

ajantang committed:
refactor<mentor> : change return value

ajantang committed:
feat : add kebab dropdown in product & comment

상품 & 댓글 삭제용 kebab dropdown 추가
테스트 과정에서 Unauthorize 이슈로 삭제 테스트 불가

참고 : 삭제/수정 코드는 구현되어 있으나 수정 관련 ui 미구현

ajantang committed:
feat : add temp redirect code

상품 상세 페이지 url을 통한 직접 접근시 무조건 sign-in으로 이동되는 현상을 해결하지 못해
임시로 기존의 경로를 저장. 저장된 경로로 로그인 성공시 이동하도록 임시 코드 작성

ajantang committed:
style : arrange css and component

중복 컴포넌트 정리
css 정리

참고 : 게시글에서 사용하던 comment list의 prop가 변경되어 추후 게시글 쪽 코드&백엔드 response 구조 변경 예정

ajantang committed:
feat : apply functions of product favorite

상품 상세 페이지의 하트를 클릭하여 favorite api 사용

ajantang committed:
style : arrange favorite text code

함수로 count가  10000 이상일 때 9999+ 텍스트로 변경

ajantang committed:
feat : validate img array of product

상품 상세 페이지의 상품 이미지 배열 유효성 검사 및 교체

ajantang committed:
fix : leak product images memory & arrange component

상품 이미지 메모리 해제 코드 추가
기존의 컴포넌트 css / 코드 정리

참고 : 아무리 생각해도 article보다 post가 더 좋은 네이밍으로 보여 교체중

ajantang committed:
feat : add delete comment function

댓글 삭제 기능 연결

ajantang committed:
feat : add eventListener for dropdown

드롭다운 영역 외 mousedown 이벤트 발생시 dropdown menu 미출력

ajantang committed:
feat : add comment modifier

댓글 수정 UI 추가
API 연결

ajantang committed:
style : replace module.css file to tailwind style

module.css 파일로 정의된 바닐라 css tailwind 스타일로 일부 변경

ajantang committed:
feat : add code product modify, delete

상품 삭제 모달 추가
상품 수정/삭제 로직 정리

ajantang committed:
test : deploy test

vercel 배포 테스트용 commit

ajantang committed:
fix : CommentList path?

배포 과정에서 문제가 생겨서 코드 재작성

ajantang committed:
feat : replace old code(not migrated) to temp code

배포를 위해 임시로 코드 수정

ajantang committed:
feat : delete sample data

배포 시도하기 위해 필요없는 코드 수정 중

ajantang committed:
fix : some warning of dependency

배포를 위한 일부 dependency 경고 처리

ajantang committed:
fix : dynamic page 404 error

동적으로 생성되는 상품 상세 페이지 404 에러

ajantang committed:
fix : dynamic page 404 error

동적으로 생성되는 상품 상세 페이지 404 에러

ajantang committed:
fix : dynamic page 404 error

동적으로 생성되는 페이지들 vercel.json 으로 설정

wildCodingWarrior commented on app/items/[productId]/DropDownKebabProduct.js:
```diff
@@ -0,0 +1,124 @@
+          )}
+        </div>
+      </dropdownContext.Provider>
+      <DeleteModal
+        showModal={showModal}
+        deleteModal={handleDeleteProduct}
+        onClose={handleHideModal}
+      />
+    </>
```
간단한 로직이라면 그냥 이런식으로 곧바로 작성해도 무방하다고 생각합니다.

다만 react에서 제공하는 createPortal 사용해도 괜찮았을 것 같네요

wildCodingWarrior commented on app/hooks/useAuth.js:
```diff
@@ -5,12 +5,26 @@ const useAuth = create((set) => ({
+        userId: user.id,
+        userNickname: user.nickname,
+        userProfileUrl: user.image,
+      };
```
로그인에 해당하는 이 부분은 
https://zustand.docs.pmnd.rs/integrations/persisting-store-data
여기 있는 미들웨어 활용하는게 사용자 경험에 훨씬 도움이 되겠습니다.

wildCodingWarrior commented on lib/axios-codeit-token.js:
```diff
@@ -0,0 +1,46 @@
+  baseURL: process.env.NEXT_PUBLIC_CODEIT_BASE_URL,
+  // withCredentials: true, // 수업 내용에 있어서 코드잇에서 설정해둔줄
+  headers: {
+    "Content-Type": "application/json",
+  },
+};
+
+export const instanceWithToken = axios.create(axiosConfig);
+
+instanceWithToken.interceptors.request.use(
+  (config) => {
+    const token = getAccessToken();
+
+    if (token) {
+      config.headers.Authorization = `Bearer ${token}`;
+    }
+
+    return config;
+  },
+  (err) => {
+    return Promise.reject(err);
+  }
+);
+
+instanceWithToken.interceptors.response.use(
+  (res) => {
+    return res;
+  },
+  async (err) => {
+    const originalRequest = err.config;
+
+    if (err.responese?.status === 401 && !originalRequest._retry) {
+      await refreshToken();
+      originalRequest._retry = true;
+      return instance(originalRequest);
+    }
+
+    return Promise.reject(err);
+  }
+);
```
활용이 아주 좋네요

wildCodingWarrior commented on vercel.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "rewrites": [
+    {
+      "source": "/items/:productId",
+      "destination": "/app/items/:productId"
+    },
+    {
+      "source": "/bulletin-board/:articleId",
+      "destination": "/app/bulletin-board/:articleId"
+    },
+    {
+      "source": "/article-edit/:articleId",
+      "destination": "/app/article-edit/:articleId"
+    }
+  ]
+}
```
404 에러가 이거랑 관련되어 있을 가능성이 높습니다. 이건 왜 있는 코드인가요?

wildCodingWarrior commented on lib/api-codeit-user.js:
```diff
@@ -1,19 +1,19 @@
+    "nickname": "닉네임",
+    "id": 1
+  }
+ */
 export async function getMyInfo() {
   const path = "/users/me";
 
-  //   {
-  //     "updatedAt": "2024-09-23T05:28:51.088Z",
-  //     "createdAt": "2024-09-23T05:28:51.088Z",
-  //     "image": "https://example.com/...",
-  //     "nickname": "닉네임",
-  //     "id": 1
-  //   }
-
   try {
-    const res = await instance.get(path);
-    console.log("getMyInfo : ", res);
+    const res = await instanceWithToken.get(path);
     return res.data;
   } catch (err) {
     throw err;
```
이 파일 이름들 굳이 api라는 이름을 파일명에 안 붙이고 폴더에 정리하는게 더 깔끔할 것 같습니다.

wildCodingWarrior commented on app/sign-up/SignUpSet.js:
```diff
@@ -39,26 +40,28 @@ export function NicknameInput({ label, register, errors }) {
             value: MIN_NICKNAME_LENGTH,
-            message: `${MIN_NICKNAME_LENGTH}자 이상 비밀번호가 필요합니다`,
+            message: `${MIN_NICKNAME_LENGTH}자 이상 닉네임가 필요합니다`,
           },
           maxLength: {
             value: MAX_NICKNAME_LENGTH,
-            message: `${MAX_NICKNAME_LENGTH}자 이상 비밀번호가 필요합니다`,
+            message: `${MAX_NICKNAME_LENGTH}자 이하 닉네임가 필요합니다`,
           },
         })}
```
오타겠죠? 닉네임가

wildCodingWarrior commented on app/items/[productId]/ProductImageListSet.js:
```diff
@@ -0,0 +1,71 @@
+    return () => {
+      setValidImgs([]);
+
+      images.forEach((img) => {
+        const image = new window.Image();
+        image.src = img;
+        image.onload = null;
+        image.onerror = null;
+      });
```
이 부분 불필요하지 않나요?

wildCodingWarrior reviewed (approved):
배포하신 주소 살펴보니 상세 페이지는 접속이 불가능하네요. 하지만 늘 그렇듯이 전체적으로 고민을 많이 하신 흔적이 보이는 코드입니다. 점점 더 코드가 깔끔해지는 것이 느껴지네요.

404 에러의 경우에는 실제로 같이 테스트를 해봐야 해결 할 수 있을 것 같은데, 멘토링 30분 진행하고 남은 시간에 같이 문제 해결해 보면 좋겠습니다.

ajantang committed:
feat : delete vercel.json data

ajantang committed:
feat : delete vercel.json file

ajantang committed:
fix : annotate bulletin-board page

중고마켓 쪽에서 사용하는 alert과 스프린트 미션 10으로 인한 BE 수정으로 prerender error
임시로 추석처리
